tardis/utilities/executors/sshexecutor.py

```diff
@@ -3,43 +3,107 @@
 from ...interfaces.executor import Executor
 from ..attributedict import AttributeDict
 
+try:
+    from contextlib import asynccontextmanager
+except ImportError:
+    from aiotools import async_ctx_manager as asynccontextmanager
+
 import asyncssh
+import asyncio
+
+
+class SSHConnectionPool(object):
+    _connection_pool = dict()
+    _lock = None
```

Comment:
This one tripped me up. It implies the lock is shared (like ``_connection_pool``), which it isn't. Took me a while to realise replacing it with an per-instance lock is intentional, not a bug.

I also think that ``_lock`` should be "as shared as" ``_connection_pool``. Either they're both global, or both per-instance. Otherwise, the lock is either too permissive or too restrictive.